Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -134,9 +137,34 @@ public final class SourceLocationConverter {
   ///   - source: The source code to convert positions to line/columns for.
   public init(file: String, source: String) {
     self.file = file
+    self.source = Array(source.utf8)
     (self.lines, endOfFile) = computeLines(source)
     assert(source.utf8.count == endOfFile.utf8Offset)
   }
+  
+  /// Execute the body with an array that contains each source line.
+  func withSourceLines<T>(_ body: ([SyntaxText]) throws -> T) rethrows -> T {
+    return try source.withUnsafeBufferPointer { (sourcePointer: UnsafeBufferPointer<UInt8>) in
+      var result: [SyntaxText] = []
+      var previousLoc = AbsolutePosition.startOfFile
```

Comment:
I know that I authored this change but reading it again, I think it would be good to add
```swift
assert(lines.first == AbsolutePosition.startOfFile)
```
here.